src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.hellbender.tools;
+
+import com.google.common.collect.Lists;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.tools.genomicsdb.GenomicsDBImport;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.test.GenomicsDBTestUtils;
+import picard.cmdline.programgroups.None;
+
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(summary = "fix the sample names in a vcf that ran through the GenotypeGVCFs pipeline when " +
+        "the batch ordering bug was present, this will restore the correct sample names provided it is given the exact sample name mapping / vcf ordering" +
+        "and batch sized that was used in the initial import",
+        oneLineSummary = "fix sample names in a shuffled callset",
+        programGroup = None.class,
+        omitFromCommandLine = true)
+public final class FixCallSetSampleOrdering extends VariantWalker {
+    @Argument(fullName = GenomicsDBImport.SAMPLE_NAME_MAP_LONG_NAME,
+            doc="the same sampleNameMap file which was used to generate the callset initially, it's important that this",
+            optional = false)
+    public String sampleNameMapPath;
+
+    @Argument(fullName = GenomicsDBImport.BATCHSIZE_ARG_NAME,
+            doc="the exact batch size that was used to generate the callset initially",
+            minValue = 1,
+            optional = false)
+    public int batchSize;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="where to write the new unshuffled VCF",
+            optional = false)
+    public File output;
+
+    private static VariantContextWriter writer;
+
+    @Override
+    public void onTraversalStart() {
```

Comment:
Another thought: when people hear about this bug they are going to panic and perhaps run this tool inappropriately. It might therefore be a good idea to print a clear description of the bug and how it was triggered on startup, then pause the tool with a "Continue? y/n" sort of prompt to give the user a chance to abort if it turns out they learn that their vcf wasn't affected based on the bug description.

Reply:
Oh, and add a warning that this repair tool will corrupt vcfs that did not actually trigger the sample ordering bug during `GenomicsDBImport`, so it's important to be 100% sure before proceeding.

Reply:
added a big warning and a prompt